tests/test_0692_fsspec_reading.py

```diff
@@ -12,6 +13,35 @@
 import sys
 
 
+@pytest.mark.parametrize(
+    "urlpath, source_class",
+    [
+        ("file.root", uproot.source.fsspec.FSSpecSource),
+        ("s3://path/file.root", uproot.source.fsspec.FSSpecSource),
+        (r"C:\path\file.root", uproot.source.fsspec.FSSpecSource),
+        (r"file://C:\path\file.root", uproot.source.fsspec.FSSpecSource),
```

Comment:
This is an example with a drive letter and a `file://`, but what about a drive letter without `file://`, or with `file:`, or `file:///`, or backward versus forward slashes? A lot of these things are considered legal (anything that could be copy-pasted into a URL bar in Edge).

Reply:
> This is an example with a drive letter and a `file://`, but what about a drive letter without `file://`, or with `file:`, or `file:///`, or backward versus forward slashes? A lot of these things are considered legal (anything that could be copy-pasted into a URL bar in Edge).

The reason I removed the test file is because we no longer do a processing of the path, we just split the object from the path if it exists, then feed the path directly to fsspec. fsspec should be smart enough to handle these different cases.

Now that I think about it this is only true if fsspec source is the one to be used, if the user manually selects some other source, we are losing this processing... I think the goal here is to let fsspec do all processing, if fsspec is not able to resolve some path and we think it should, we should open an issue in fsspec. We are only responsible of transforming the non-compliant urlpath:object into a compliant urlpath (that's my idea atleast).

With this approach things such as `file:///g:/mydir/file.root` are not going to work. They would get interpreted as `/g:/mydir/file.root` which is not a valid windows path... (I think) should they work, though? My browser also interprets it as `/g:/mydir/file.root`. However on issue https://github.com/scikit-hep/uproot5/issues/325 it's stated as a valid windows path.

Also the previous implementation treats non-schema paths and paths with `file://` differently, which I don't really understand. fsspec treats paths without schema as local filesystem paths: `fsspec.core.url_to_fs("file:///tmp/file.root")` and `fsspec.core.url_to_fs("/tmp/file.root")` are the same. But different from `fsspec.core.url_to_fs("file://tmp/file.root")` which assumes `tmp/file.root` is a non-absolute path, same as `fsspec.core.url_to_fs("file:tmp/file.root")` or `fsspec.core.url_to_fs("tmp/file.root")`.

My point is: should uproot be smarter than fsspec when resolving paths? assuming the user passes a filepath (without specifying an object in the uproot-way), we should just pass it straight to fsspec without doing anything to it. If fsspec fails to resolve it, then the path is wrong or it isn't and the issue should be raised in fsspec, not uproot. Do you agree with this @jpivarski ? I think this is in line with the motivation of the fsspec integration: offload responsability from uproot.

Reply:
Uproot should not be smarter than fsspec about resolving paths. That is fsspec's job. We should at least know what fsspec does, though. It may be that by handing this over to fsspec, some of our old tests no longer pass, and that's something that we can note in the test code as a reminder if it ever comes up. (This is something that users will bring up to us, since we don't use Windows, but some of them do.)

If fsspec doesn't accept these paths, we can also ask @martindurant why not. Maybe it hasn't been considered, or maybe `file:g:\file.root` is not satisfying any protocol and Uproot had been too loose in accepting it.

Reply:
> maybe file:g:\file.root is not satisfying any protocol

I think it would have to be
file://g:/file.root or file://g:\file.root or g:/file.root

I've only ever seen the single colon in terms of windows drive letters and the special case of data: URLs. fsspec considers protocols to be the thing before "::" (separating multiple protocol parts) or "://".

Reply:
I agree.

To be clear: `file:g:\file.root` is treated as `file://g:\file.root` (`g:\file.root` without schema) by fsspec and is thus a valid path. This would still work.

`file:/g:\file.root` is treated as `file:///g:\file.root` (`/g:\file.root` without schema) which is not a valid windows path (according to https://github.com/scikit-hep/uproot5/issues/325). These are currently supported by uproot but wouldn't be after this PR if merged as of right now.

Reply:
We can lose some capabilities if we had just been too loose with it in the past. All I was asking for here was to know that we're losing them and why.

I had been taking "a web browser would accept it" as a way of defining good URIs. I think web browsers accept `file:path/to/file` (only for local paths, not `http:path/to/file`), though I can't personally test how this would interact with Windows drive letters.

So it may be that all this PR needs to be complete is an accounting of which patterns we now accept and which have been lost, on purpose.